### PR TITLE
Fix some request loops, allow message sending

### PR DESCRIPTION
**Pull request by mikeger**
# Issue with past participants

Due to the change in the backend, the `/conversations/{id}` is permanently returning 404 for requests from the users that are not the part of the conversation any more.

Solution is to mark such conversations with `needsToBeUpdatedFromBackend = false`, same applies for users, teams and connections.

# Issue with multiple `UserClient`

Some users accumulated the big amount of duplicated* user clients in the databases due to the unknown bug. 

Solution is to merge the user clients for now and monitor if the issue would continue to happen in the future, since the unknown bug might be fixed by the previous attempts.

* Duplicated clients are the clients with the exact same user and remoteIdentifier inserted in the database multiple times.

# Remarks

- `DiskDatabaseTest` is added since the CoreData's `GROUP BY` is only working on the disk storage.

# Depends on

- [x] https://github.com/wireapp/wire-ios-message-strategy/pull/81
- [x] https://github.com/wireapp/wire-ios-data-model/pull/293
- [x] https://github.com/wireapp/wire-ios-data-model/pull/299

### CODE REVIEW

**Comment by mikeger** [comment]
@basine good point, would add!

BTW currently blocked by changes in https://github.com/wireapp/wire-ios-sync-engine/pull/418

**Comment by mikeger** [comment]
@basine updated, please re-review!

**Comment by mikeger** [comment]
Tests are breaking, needs some fixing ❌

**Comment by mikeger** [comment]
@basine @daehn please re-review

File: Source/Synchronization/ZMHotFixDirectory+Swift.swift [hunk 1]
```diff
@@ -110,4 +132,24 @@ extension ZMHotFixDirectory {
...
+        context.findDuplicated(by: #keyPath(UserClient.remoteIdentifier)).forEach { (remoteId: String?, clients: [UserClient]) in
+            // Group clients having the same remote identifiers by user
+            clients.filter { !($0.user?.isSelfUser ?? true) }.group(by: ZMUserClientUserKey).forEach { (user: ZMUser, clients: [UserClient]) in
+                guard let firstClient = clients.first, clients.count > 1 else {
+                    return
+                }
+                
+                let tail = clients.dropFirst()
+                // Merge clients having the same remote identifier and same user
+                let _ = tail.reduce(firstClient) {
```
Comment (daehn) [inline Source/Synchronization/ZMHotFixDirectory+Swift.swift]: I would avoid using `reduce` to perform side effects. I guess the same result could be achieved using a simple `for` loop?

**Comment by daehn** [comment]
@mikeger conflicting 👻

**Comment by mikeger** [comment]
Can be merged now.

File: WireSyncEngine.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2332,7 +2332,6 @@
 				5447E4631AECDC2E00411FCD /* ZMUserSessionRegistrationTests.m in Sources */,
 				543ED0011D79E0EE00A9CDF3 /* ApplicationMock.swift in Sources */,
 				878F63E51EF91E7C006E819B /* HotFixDuplicatesTests.swift in Sources */,
-				BF1F52C81ECC8720002FB553 /* TeamSyncRequestStrategyTests.swift in Sources */,
```
Comment (daehn) [inline WireSyncEngine.xcodeproj/project.pbxproj]: Not sure, this one should have been re-added with @basine 's latest changes.

Comment (mikeger) [inline WireSyncEngine.xcodeproj/project.pbxproj]: May be but check the final diff, I think it was added at some point but then removed.

**Pull request merged**